packages/playwright-test/src/dispatcher.ts

```diff
@@ -215,7 +215,7 @@ export class Dispatcher {
       }));
       result.status = params.status;
       test.expectedStatus = params.expectedStatus;
-      test.annotations = params.annotations;
+      test._annotateWithInheritence(params.annotations);
```

Comment:
Although the logic has been moved out of the dispatcher, this still smells. However, since we depend on round-tripping the annotations, I think—short of refactoring how annotations are plumbed through—this may be a decent fix for now.

In the future, I think it would make sense to solve the issue from the other direction.